api/v1alpha7/openstackmachine_types.go

```diff
@@ -84,6 +84,9 @@ type OpenStackMachineSpec struct {
 	// The volume metadata to boot from
 	RootVolume *RootVolume `json:"rootVolume,omitempty"`
 
+	// Data volumes to attach to the server instance
```

Comment:
I know we don't even remotely follow this in the existing APIs, but for new APIs we should follow the convention that the first word of the field description is the exact name of the field.

So something like
```golang
// DataVolumes defines a list of volumes to attach to the server instance
```